src/Client/PartnerizeClient.php

```diff
@@ -107,11 +111,59 @@ public function approveConversion(string $conversionId): void
      * @param string $conversionId
      * @param string $reason
      *
+     * @return Job
      * @throws ClientException
      */
-    public function rejectConversion(string $conversionId, string $reason): void
+    public function rejectConversion(string $conversionId, string $reason): Job
     {
-        $this->setConversionStatus($conversionId, self::STATUS_REJECTED);
+        return $this->setConversionStatus($conversionId, self::STATUS_REJECTED);
+    }
+
+    /**
+     * @param string $id
+     *
+     * @return Job
+     * @throws ClientException
+     */
+    public function getJobUpdate(string $id): Job
+    {
+        try {
+            $response = $this->apiClient->request(
+                'GET',
+                sprintf('job/%s', $id)
+            );
+        } catch (GuzzleException $exception) {
+            throw new ClientException($exception->getMessage(), 0, $exception);
+        }
+
+        return $this->deserializeJob($response);
+    }
+
+    /**
+     * @param string $id
+     *
+     * @return Response
+     * @throws ClientException
+     */
+    public function getJobResponse(string $id): Response
+    {
+        try {
+            $response = $this->apiClient->request(
+                'GET',
+                sprintf('job/%s/response', $id)
+            );
+        } catch (GuzzleException $exception) {
+            throw new ClientException($exception->getMessage(), 0, $exception);
+        }
+
+        $this->checkStatusCode($response);
+
+        $json = $response->getBody()->getContents();
+
+        /** @var Response $response */
+        $response = $this->serializer->deserialize($json, Response::class, 'json');
+
+        return $response;
```

Comment:
Moet dit niet zijn:

return $this->deserializeJob($response);

Reply:
No because the method should return the Response of the Job after it was finished. You can check conversion items or errors from the Response.

Reply:
Okay does it make more sense if you refactor the functions who use `deserializeJob` to use a method named `getResponse()` and every method that requires the job can call the `$response->getJob()` respectfully.